app/controllers/steps/base_controller.rb

```diff
@@ -66,7 +66,7 @@ def initialize_commodity_context_service
     end
 
     def title
-      t("page_titles.#{@step.class.try(:id)}", default: super)
+      t("page_titles.#{@step.class.try(:id)}", uk_internal_market_scheme:, default: super)
```

Comment:
I'm wondering about coupling the base controller with a concrete title setting. Is it better to just override the title in the underlying controller?

Reply:
Potentially although i feel it depends how many options are passed into the title tag maybe? Having these sorts of changes i think will always feel a little dirty and quick because of the purpose they are solving. I guess if we have a whole heap of variables being passed in for different steps and their titles then would make sense to move the title in that controller class.

Reply:
I'll have a look at the code a bit more and see if I'm missing something but this is basically breaching SOLID principles

Reply:
Yeah, all we need to do to avoid mixing special titles is to update the trader scheme controller:

```ruby
module Steps
  class TraderSchemeController < BaseController
    helper_method :title

    def show
      @step = Steps::TraderScheme.new
    end

    def create
      @step = Steps::TraderScheme.new(permitted_params)

      validate(@step)
    end

    def title
      t("page_titles.#{@step.class.try(:id)}", uk_internal_market_scheme:, default: super)
    end

    private

    def permitted_params
      params.require(:steps_trader_scheme).permit(
        :trader_scheme,
      )
    end
  end
end
```